cypress/integration/distroSettings/host_section.ts

```diff
@@ -57,4 +57,55 @@ describe("host section", () => {
       cy.validateToast("success");
     });
   });
+
+  describe("using User Data bootstrap method", () => {
+    beforeEach(() => {
+      cy.visit("/distro/ubuntu1604-parent/settings/host");
+      cy.selectLGOption("Host Bootstrap Method", "User Data");
+      cy.selectLGOption("Host Communication Method", "RPC");
+    });
+
+    it("shows Windows-only fields when the architecture is updated", () => {
+      cy.contains("label", "Root Directory").should("not.exist");
+      cy.contains("label", "Service User").should("not.exist");
+
+      cy.selectLGOption("Agent Architecture", "Windows 64-bit");
+
+      cy.getInputByLabel("Root Directory").should("exist");
+      cy.getInputByLabel("Service User").should("exist");
+    });
+
+    it("hides resource limit fields when the architecture is not Linux", () => {
+      cy.contains("Resource Limits").should("exist");
+      cy.selectLGOption("Agent Architecture", "Windows 64-bit");
+      cy.contains("Resource Limits").should("not.exist");
+    });
+
+    it("saves bootstrap settings", () => {
+      cy.getInputByLabel("Jasper Binary Directory").type("/jasper/binary");
+      cy.getInputByLabel("Jasper Credentials Path").type("/jasper/credentials");
+      cy.getInputByLabel("Client Directory").type("/client/dir");
+      cy.getInputByLabel("Shell Path").type("/shell/path");
+      cy.getInputByLabel("Home Volume Format Command").type(
+        "echo 'Hello World'"
+      );
+      cy.getInputByLabel("Number of Files").type("10");
+      cy.getInputByLabel("Number of CGroup Tasks").type("20");
+      cy.getInputByLabel("Number of Processes").type("30");
+      cy.getInputByLabel("Locked Memory").type("128");
+      cy.getInputByLabel("Virtual Memory").type("256");
+
+      cy.contains("button", "Add variable").click();
+      cy.getInputByLabel("Key").type("my-key");
+      cy.getInputByLabel("Value").type("my-value");
+
+      cy.contains("button", "Add script").click();
+      cy.getInputByLabel(/^Path$/).type("/path/to/precondition/script");
+      cy.getInputByLabel(/^Script$/).type("script contents here");
```

Comment:
These inputs refer to the precondition script fields; the regex is actually used to make them _more_ specific so that we target the "Script" input instead of "Setup script" 
<img width="510" alt="image" src="https://github.com/evergreen-ci/spruce/assets/9298431/65a24e16-3429-4d7c-8afc-d3a0cf0f0b24">

Using a regex is probably the best way to achieve this since `getInputByLabel` uses `.contains()`